pkg/rbd/nodeserver.go

```diff
@@ -230,15 +231,16 @@ func (ns *NodeServer) stageTransaction(ctx context.Context, req *csi.NodeStageVo
 	transaction.isStagePathCreated = true
 
 	// nodeStage Path
-	err = ns.mountVolumeToStagePath(ctx, req, staticVol, stagingTargetPath, devicePath)
+	readOnly, err = ns.mountVolumeToStagePath(ctx, req, staticVol, stagingTargetPath, devicePath)
```

Comment:
I think the staging step should be writable, even when the PVC is used with `readOnly: true`. The bind-mount should be read-only. This allows `chmod` from continuing and setting the correct (expected) permissions.

Only in the multi-attach (like RWX and ROX) case, the permissions should not be changed.

Reply:
this is the readOnly option set in storageclass not the readOnly of the application pod. if the readOnly flag is set in the storageclass why we need to make the staginpath as the writable?

Reply:
Ah, ok! In that case it makes sense. (Although I do not see how a readOnly option in the storageclass can be useful.)